pkg/volume/util/metrics.go

```diff
@@ -63,6 +63,13 @@ var storageOperationEndToEndLatencyMetric = prometheus.NewHistogramVec(
 	[]string{"plugin_name", "operation_name"},
 )
 
+var volumeOperationErrorsMetric = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Name: "volume_operation_total_errors",
+		Help: "Total volume operation erros",
```

Comment:
Was this a previous metric that existed before and was being consumed?

The general feedback I've gotten from sig-instrumentation is that total errors is not useful without a total count.  So I'm not sure if we want to continue to support this metric. We currently have 3 different metrics that report errors, and I would like to deprecate 2 of them and only continue to support 1 going forward.

Reply:
No it does not. The metric, "volume_operation_total_errors" and "volume_operation_total_seconds" only exists for "provision" and "deletion" operations. For "attach" and "detach", it's newly introduced.
It's a very legit feedback. The idea is once "volume_operation_total_seconds" have been fully implemented to cover all volume operations, mount/unmount, resizing, snapshot etc, it should replace the current "storage_operation_seconds".
For the error one, it's a bit tricky, there are two things to keep in mind, one is naming, if we have decided to use "volume_operation_total_seconds", it makes sense to have the error metric named as volume_operation_xxx_errors than re-using current ones. The other would be the meaning of total errors. The retry logic in controllers makes total error count a bit different to a typical service failure call. It's totally possible the total error count being larger than total count of operations (assuming total count means total number of operations). In this case, having total error counts only seems to be not a horrible idea.

Reply:
Maybe we should take a step back and try to come up with a clear long-term model of how we want the volume metrics to look like.  For me, the picture is very confusing right now because:
* we have K8s style reconciler loops which don't quite match up with traditional service operations. 
* csi and in-tree are different
* we're trying to derive metrics for all these on the client-side, which maybe doesn't make sense when considering csi

Reply:
Conclusion from offline discussion is let's just focus on e2e latency for now.  We will followup with a future proposal/KEP on how to count errors at various levels: plugin, controller, user.

Reply:
Also, there's a typo in the help msg, should be "errors"

Reply:
fixed. thanks